compiler/src/typed/datarepr.re

```diff
@@ -141,7 +137,11 @@ let extension_descr = (path_ext, ext) => {
   let ty_res = newgenconstr(ext.ext_type_path, ext.ext_type_params);
 
   let (existentials, cstr_args, cstr_inlined) =
-    constructor_args(ext.ext_args, Some(ty_res), path_ext);
+    constructor_args(
+      ext.ext_args,
+      Some(ty_res),
+      Path.PExternal(path_ext, "#extension#"),
```

Comment:
Kind of a hack; the OCaml compiler extracts "extension" constructors into a separate variant but doing this would require changing a lot of other compiler code so I just use this string as a unique identifier for exceptions (but lmk if yall think it's still worth it to create a new variant)